ibtk/src/lagrangian/LEInteractor.cpp

```diff
@@ -1311,6 +1311,108 @@ spread_data(const int stencil_sz,
     }
 #endif
 } // spread_data
+
+// For internal use - convert to an enumeration. This may make it into the
+// public interface one day but for backwards compatibility we accept only
+// strings for now.
+enum KernelName
+{
+    BSPLINE_3,
+    BSPLINE_4,
+    BSPLINE_5,
+    BSPLINE_6,
+    DISCONTINUOUS_LINEAR,
+    PIECEWISE_CONSTANT,
+    PIECEWISE_LINEAR,
+    PIECEWISE_CUBIC,
+    IB_3,
+    IB_4,
+    IB_4_W8,
+    IB_5,
+    IB_6,
+    USER_DEFINED,
+    INVALID
+};
+
+KernelName
+string_to_kernel(const std::string& kernel_fcn)
+{
+    // Some IBAMR applications want to call this function *a lot*, e.g., ideally
+    // we would call this on every element in a finite element mesh instead of
+    // batching with huge temporary arrays. Hence doing all these string
+    // comparisons can get surprisingly expensive. Chop them up first with a
+    // switch statement.
+#ifndef NDEBUG
+    if (!LEInteractor::isKnownKernel(kernel_fcn)) goto unknown_kernel;
```

Comment:
`goto`??? 😁

Reply:
It's a reasonable way to handle failure in this case!

I try not to write more than one per decade.